lib/pure/logging.nim

```diff
@@ -503,12 +508,8 @@ when not defined(js):
 
   # ------
 
-  proc countLogLines(logger: RollingFileLogger): int =
-    result = 0
-    let fp = open(logger.baseName, fmRead)
-    for line in fp.lines():
-      result.inc()
-    fp.close()
+  template countLogLines(logger: RollingFileLogger): int =
+    readFile(logger.baseName).countLines()
```

Comment:
Hm, doesn't this read in the entire file though? That might cause some pretty large allocations in some circumstances.